horovod/tensorflow/mpi_ops.cc

```diff
@@ -248,11 +276,47 @@ TFOpContext::AllocateOutput(common::TensorShape shape,
   return ConvertStatus(status);
 }
 
+int GetDeviceID(OpKernelContext* context);
+
 common::Status
 TFOpContext::AllocateZeros(int64_t num_elements, common::DataType dtype,
                            std::shared_ptr<common::Tensor>* tensor) {
-  return common::Status::PreconditionError(
-      "AllocateZeros is not supported for TensorFlow yet.");
+  ::tensorflow::Tensor* unused;
+  std::shared_ptr<PersistentTensor> zero_tensor = std::make_shared<PersistentTensor>();
+  auto tf_data_type = GetTFDataType(dtype);
+  ::tensorflow::AllocatorAttributes tf_attribute;
+  int device_ = GetDeviceID(context_); 
+  auto hvd_context = std::make_shared<TFOpContext>(context_);
+  if (device_ != CPU_DEVICE_ID){
+    tf_attribute.set_on_host(false);
+  } else{
+    tf_attribute.set_on_host(true);
+  }
+
+  Status status = context_->allocate_persistent(tf_data_type, ::tensorflow::TensorShape({num_elements}), zero_tensor.get(), &unused, tf_attribute);
+   
+  if (device_ != CPU_DEVICE_ID){
```

Comment:
Please add space after if conditions and opening bracket in this file like this: `if (device_ != CPU_DEVICE_ID) {`

Same applies to else. Should be `} else {`.